### PR TITLE
fix(amazon/deploy): Do not destroy SpEL based load balancers in deploy config

### PR DESCRIPTION
**Pull request by jrsquared**
If someone has added a SpEL based load balancer or target group to a deploy config, then goes to edit that deploy config in the UI, it will remove them because they don't string match existing ones.

This basically trusts load balancers or target groups that have a `${` in them as being 'valid' so it will not remove them.

### CODE REVIEW

File: app/scripts/modules/amazon/src/serverGroup/configure/wizard/loadBalancers/loadBalancerSelector.component.html [hunk 1]
```diff
@@ -26,7 +26,7 @@
                  ng-model="$ctrl.command.targetGroups"
                  class="form-control input-sm">
         <ui-select-match>{{$item}}</ui-select-match>
-        <ui-select-choices repeat="targetGroup in $ctrl.command.backingData.filtered.targetGroups | filter: $select.search">
+        <ui-select-choices repeat="targetGroup in $ctrl.command.backingData.filtered.targetGroups.concat($ctrl.command.spelTargetGroups) | filter: $select.search">
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/serverGroup/configure/wizard/loadBalancers/loadBalancerSelector.component.html]: pretty sure I'm angularing wrong with a `.concat` in here...

**Comment by anotherchrisberry** [comment]
I've been doing something a little simpler lately when people run into this: if `${` is part of the value, show a plain text input instead of the select box. That lets people edit the value in the dialog without editing the JSON (once they've edited the JSON initially)

**Comment by anotherchrisberry** [comment]
Added a line item in the refactor action items list for "SpEL support for all pipeline fields". This will be a good one to get figured out pretty quickly when we start converting pipeline configuration to React.

**Comment by anotherchrisberry** [comment]
(LGTM)

**Pull request merged**